src/Shared/ServerInfrastructure/HttpCharacters.cs

```diff
@@ -1,148 +1,98 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics;
 using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
 
 namespace Microsoft.AspNetCore.Http;
 
-internal static class HttpCharacters
+[SkipLocalsInit]
+internal static unsafe partial class HttpCharacters
```

Comment:
I developed this with a playground-project in https://github.com/gfoidl-Tests/SourceGenerator-Vectors

For best codegen, a code-generator is used (in the project it's still the non-incremental one, I didn't update that code).

[Methods in the generator](https://github.com/gfoidl-Tests/SourceGenerator-Vectors/blob/3e9ef63ed834f5218fae4b9e073ac09c78f6b917/ConsoleApp2.Generator/HttpCharacters_Vectorized.Generator.cs#L45-L47) do "the same" as was done in the static init before.

Thus there's no need for static initialization at runtime, instead 
* `ROS<byte>` -- will refer to the assembly's static data segment -- can be used
* for vectors it's best to use constants like `Vector128.Create((byte)0xF)` inline instead of loading the from `static readonly` fields

In this draft-PR I just copied the output from the source-generator over to here (see file below).

What's the right strategy to generate such a file?

A source-generator, tt-file, etc. Please advise so I can update the PR here to make it suite.